bigquery/google/cloud/bigquery/client.py

```diff
@@ -2376,8 +2376,11 @@ def insert_rows(self, table, rows, selected_fields=None, **kwargs):
                 the mappings describing one or more problems with the row.
 
         Raises:
-            ValueError: if table's schema is not set
+            ValueError: if table's schema is not set or `rows` is not a `Sequence`.
         """
+        if not isinstance(rows, (collections_abc.Sequence, collections_abc.Iterator)):
+            raise TypeError("rows argument should be a sequence of dicts or tuples")
```

Comment:
Adding `Iterator` too as `insert_rows_from_dataframe()` calls `insert_rows()` with iterator returned by `itertools.islice`. `dict` still will give a `False` at this statement